### ISSUE TITLE
Popover: Pass aria-haspopup to triggering component

### ISSUE THREAD

**Issue by asha-cs** [salesforce/design-system-react#2302]
* The Popover component should provide a way to supply a value for the `aria-haspopup` property. This will help customers using screen reader technology to be aware that the button will open other content in a popup.
* Salesforce bug: W-6564300
* I am not currently planning on fixing this issue.

**Comment by interactivellama** [comment]
The fix should be to pass an `aria-haspopup='dialog'` down to the triggering component automatically, since the Popover is the parent. Right now, the prop would need to be added manually by the consumer.
